src/core/data_types.rs

```diff
@@ -527,4 +545,46 @@ mod tests {
         assert_eq!(r.cycle_focus(Direction::Backward), Some(&1));
         assert_eq!(r.as_vec(), vec![1, 2, 3]);
     }
+
+    #[test]
+    fn element() {
+        let mut r = Ring::new(vec![1, 2, 3, 4]);
+        assert_eq!(r.element(&Selector::Condition(&|e| e % 2 == 0)), Some(&2));
+        assert_eq!(r.element_mut(&Selector::Condition(&|e| e % 2 == 0)), Some(&mut 2));
+
+        assert_eq!(r.element(&Selector::Index(2)), Some(&3));
+        assert_eq!(r.element_mut(&Selector::Index(2)), Some(&mut 3));
+        assert_eq!(r.element(&Selector::Index(50)), None);
+        assert_eq!(r.element_mut(&Selector::Index(50)), None);
+
+        r.focus(&Selector::Index(1));
+        assert_eq!(r.element(&Selector::Focused), Some(&2));
+        assert_eq!(r.element_mut(&Selector::Focused), Some(&mut 2));
+
+        assert_eq!(r.element(&Selector::WinId(42)), None);
+        assert_eq!(r.element_mut(&Selector::WinId(69)), None);
+
+        assert_eq!(r.as_vec(), vec![1, 2, 3, 4]);
+    }
+
+    #[test]
+    fn all_elements() {
+        let mut r = Ring::new(vec![1, 2, 3, 4]);
+        assert_eq!(r.all_elements(&Selector::Condition(&|e| e % 2 == 0)), vec![&2, &4]);
+        assert_eq!(r.all_elements_mut(&Selector::Condition(&|e| e % 2 == 0)), vec![&mut 2, &mut 4]);
+
+        assert_eq!(r.all_elements(&Selector::Index(2)), vec![&3]);
+        assert_eq!(r.all_elements_mut(&Selector::Index(2)), vec![&mut 3]);
+        assert_eq!(r.all_elements(&Selector::Index(50)), vec![&0; 0]);
```

Comment:
Not sure if you prefer `Vec::<&i32>::new()` over `vec![&0,0]`. I can change it if you like.

Reply:
Yeah, `Vec::<&i32>::new()` is a little clearer with the intent that it's an empty vec :+1:  (or I think `vec![]` should work as well?)

Reply:
vec![] didn't work, because it could not infer the type. I'll change it tomorrow, it's getting a bit late for me.